protostar/cheatable_starknet/cheaters/expects.py

```diff
@@ -0,0 +1,104 @@
+import os
+from typing import Optional, Callable, TYPE_CHECKING
+
+from starkware.starknet.public.abi import get_selector_from_name
+
+from protostar.starknet.cheater import CheaterException
+from protostar.starknet.address import Address
+from protostar.starknet.data_transformer import (
+    CairoOrPythonData,
+    transform_calldata_to_cairo_data,
+    CairoData,
+)
+from protostar.starknet.types import SelectorType
+
+
+if TYPE_CHECKING:
+    from protostar.cheatable_starknet.cheatable_cached_state import CheatableCachedState
+
+
+def assert_no_expected_calls(
+    expected_calls: dict[Address, list[tuple[SelectorType, CairoOrPythonData]]],
+    fixed_address: Optional[Address] = None,
+) -> None:
+    msg = "expected calls not fulfilled:"
+    any_addresses_present = False
+    for address, details in expected_calls.items():
+        if fixed_address is not None and address != fixed_address:
+            continue
+        any_addresses_present = True
+        msg += f"{os.linesep}  - for address {address}:"
+        for details_item in details:
+            msg += f"{os.linesep}    - function name: {details_item[0]}, calldata: {details_item[1]}"
+    if any_addresses_present:
+        raise CheaterException(msg)
```

Comment:
We raise a `CheaterException` here, let's make this a `staticmethod` of `ExpectsCairoCheater`.

Reply:
[address feeback](https://github.com/software-mansion/protostar/pull/1363/commits/18139ec67977c1e8a2a28880088e9ba1d613f633)